src/test/xmeta/test_xmeta_emit.cpp

```diff
@@ -210,31 +263,34 @@ void test_delegate_type_properties(TypeDef const& delegate_type)
 
 std::vector<uint8_t> run_and_save_to_memory(std::istringstream & test_idl, std::string_view assembly_name)
 {
-    xmeta_idl_reader reader{ "" };
+    std::vector<std::string> paths = { "Foundation.xmeta" };
+    xmeta_idl_reader reader{ assembly_name, paths };
     reader.read(test_idl);
     REQUIRE(reader.get_num_syntax_errors() == 0);
-    xlang::xmeta::xmeta_emit emitter(assembly_name);
-    xlang::xmeta::xlang_model_walker walker(reader.get_namespaces(), emitter);
-
-    walker.register_listener(emitter);
-    walker.walk();
-
-    xlang::meta::writer::pe_writer writer;
-    writer.add_metadata(emitter.save_to_memory());
+    REQUIRE(reader.get_num_semantic_errors() == 0);
+    return reader.save_to_memory();
+}
 
-    return writer.save_to_memory();
+template<typename T>
+auto find_type_by_name(table<T> const& type, std::string name, std::string namespace_name)
```

Comment:
If this is coming up frequently, it might be worth considering `reader::cache` which will store these name-based lookups more efficiently.

Reply:
That would require alot of changes to the code and this is just a test so the performance isn't a worry.